uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/staff-leave-report.component.js

```diff
@@ -534,6 +573,7 @@ define([
         loadEntitlements(),
         loadBalanceChanges()
       ])
+        .then(processAbsenceTypes)
```

Comment:
even if this is used in two different places, I rather repeat this:

```js
// ...
  .then(attachEntitlementsToAbsenceTypes)
  .then(filterAbsenceTypes)
// ...
```
The reason being that I have no idea what `processAbsenceTypes`, but `attachEntitlementsToAbsenceTypes` and `filterAbsenceTypes` sound more concrete.

what do you think?

Reply:
I was thinking about that, but there is also the argument that one can have that it would be better to handle anything related to the processing (enhance + filter, in this case) of absence types in a single wrapping function. Otherwise it could look like that `attachEntitlementsToAbsenceTypes`and  `filterAbsenceTypes` could be called independently from each other, but that's not the case (you can filter only after you attach)